docs/opensafely-cli.md

```diff
@@ -117,15 +117,75 @@ See the [Codelist](codelist-intro.md) section for more information on codelists.
 
 
 To run your code on your machine, the `opensafely` tool uses the same Docker
-images that run in the secure server environments. These are updated
-periodically, for example when new libraries are installed. If you have error
-messages about missing libraries, your Docker images may need upgrading.
-To pull the most recent Docker images to your machine, run:
+images that run in the secure server environments. There is the
+`cohortextractor` image, for processing study definitions, and then `r`,
+`stata-mp`, and `python` images, for running your analysis code. These last
+three provide a pre-built environment for their specific tool, with a set of libraries.
+
+
+These are updated periodically, for example when new libraries are installed.
+If you have error messages about missing libraries, your Docker images may need
+upgrading.  To pull the most recent Docker images to your machine, run:
 
 ```bash
 opensafely pull
 ```
 
+
+### Developing code with the OpenSAFELY images.
+
+The R and Stata tools are often used interactively to develop and test analysis code, and python users often use `ipython` or `jupyterlab` to develop and test code.
+
+The `opensafely exec` command provides a simple way to do this, by running interactive tools using the Docker images provided by OpenSAFELY. This can help ensure that your code works correctly in OpenSAFELY as you develop it, rather than accidentally relying on tools and libraries installed on your own machine.
+
+Running `opensafely exec IMAGE COMMAND` does the following:
+ - runs an instance of the appropriate docker IMAGE (`r`, `python`, `stata-mp`)
+ - shares the files in your current directory with the instance
+ - executes COMMAND (or the default command for the image if you don't supply one)
+
+This allows you develop and and test code as if it was a regular interactive session.
+
+For example, to run an interactive Stata session:
+
+```bash
+opensafely exec stata-mp
+```
+
+This will run the Stata packaged in the `stata-mp` docker image, and you can manually test your Stata code. (the opensafely tool knows how to fetch and apply the OpenSAFELY Stata licence).
+
+
+Likewise, for R:
+
+```bash
+opensafely exec r R
```

Comment:
Note: this won't work until the new R image is out, but that's hopefully soon.

Reply:
this is where we could link to a future R demonstration